test/commands/scanner/run.dfa.test.ts

```diff
@@ -0,0 +1,81 @@
+import {expect} from '@salesforce/command/lib/test';
+import {setupCommandTest} from '../../TestUtils';
+import {Messages} from '@salesforce/core';
+import * as path from 'path';
+
+
+Messages.importMessagesDirectory(__dirname);
+const sfgeMessages = Messages.loadMessages('@salesforce/sfdx-scanner', 'EventKeyTemplates');
+
+const dfaTarget = path.join('test', 'code-fixtures', 'projects', 'sfge-smoke-app', 'src');
+const projectdir = path.join('test', 'code-fixtures', 'projects', 'sfge-smoke-app', 'src');
+
+const apexControllerStr = 'UnsafeVfController';
+const customSettingsStr = 'none found';
+const fileCount = '7';
+const entryPointCount = '5';
+const pathCount = '6';
+const violationCount = '2';
+
+const customSettingsMessage = sfgeMessages.getMessage('info.sfgeMetaInfoCollected', ['Custom Settings', customSettingsStr]);
+const apexControllerMessage = sfgeMessages.getMessage('info.sfgeMetaInfoCollected', ['Apex Controllers', apexControllerStr]);
+const compiledMessage = sfgeMessages.getMessage('info.sfgeFinishedCompilingFiles', [fileCount]);
+const startGraphBuildMessage = sfgeMessages.getMessage('info.sfgeStartedBuildingGraph');
+const endGraphBuildMessage = sfgeMessages.getMessage('info.sfgeFinishedBuildingGraph');
+const identifiedEntryMessage = sfgeMessages.getMessage('info.sfgePathEntryPointsIdentified', [entryPointCount]);
+const completedAnalysisMessage = sfgeMessages.getMessage('info.sfgeCompletedPathAnalysis', [pathCount, entryPointCount, violationCount]);
+
+function isSubstr(output: string, substring: string): boolean {
+	const regex = new RegExp(`^${substring}`, 'gm');
+	const regexMatch = output.match(regex);
+	return regexMatch != null && regexMatch.length >= 1;
+}
+
+function verifyContains(output: string, substring: string): void {
+	expect(isSubstr(output, substring), `Output "${output}" should contain substring "${substring}"`).is.true;
+}
+
+function verifyNotContains(output: string, substring: string): void {
+	expect(isSubstr(output, substring), `Output "${output}" should not contain substring "${substring}"`).is.false;
+}
+
+describe('scanner:run:dfa', function () {
```

Comment:
New end-to-end tests for dfa. Directly testing sfge's Main.java was getting problematic because of the multithreading. These new tests are more reliable.
Though I'm seeing an issue where each verbose like is printed 6 times. Still investigating to understand the cause.